ghost/core/core/server/services/members/middleware.js

```diff
@@ -156,14 +156,7 @@ const createSessionFromMagicLink = async function (req, res, next) {
 
     try {
         const member = await membersService.ssr.exchangeTokenForSession(req, res);
-        await new Promise((resolve, reject) => {
-            spamPrevention.membersAuth().reset(req.ip, `${member.email}login`, function (err) {
-                if (err) {
-                    reject(err);
-                }
-                resolve();
-            });
-        });
+        spamPrevention.membersAuth().reset(req.ip, `${member.email}login`);
```

Comment:
We don't actually want to stop someone from signing in if this fails, and we also don't want to block the request on it - so running this in the background is preferred